### PR TITLE
Add filewatcher to enable custom config support for LCLS

### LINKED ISSUE #490: Change file monitoring for LCLS
To support [custom config files](https://github.com/OpenLiberty/liberty-language-server/issues/66), after OpenLiberty/liberty-language-server#212 goes in, the IDE client side needs to be changed to monitor the following to send to LCLS:
- all `.env` and `.properties` files - language support
- `liberty-plugin-config.xml` - watch file for changes
  - since IJ doesn't currently have built-in LSP support, may have to add our own [Listener](https://plugins.jetbrains.com/docs/intellij/plugin-listeners.html) or File Watcher implementation and link it to the [LibertyWorkspaceService](https://github.com/OpenLiberty/liberty-language-server/blob/a1b25f2db7659335b71a50e368c8cb0c3673938b/liberty-ls/src/main/java/io/openliberty/tools/langserver/LibertyWorkspaceService.java#L36-L44)

https://github.com/OpenLiberty/liberty-tools-intellij/blob/b66def71cd40456e2f36db7adb930f2ff18b4e41/src/main/resources/META-INF/lsp.xml#L39-L40

Whether or not this change is implemented shouldn't break anything, but will not have support for custom config files.

Can model after OpenLiberty/liberty-tools-vscode#286

### PR DESCRIPTION
**Pull request by evie-lau**
Resolves #490 
Add file listener - notify LCLS to process `liberty-plugin-config.xml` files when a filechange is detected.

Also requires LCLS 2.1 or 2.1-SNAPSHOT
Merging this change does not break existing support for LCLS files.

## Notes
Currently this only works if `liberty-plugin-config.xml` changes _**after**_ IntelliJ/LCLS is initialized. 
Liberty Tools IntelliJ doesn't initialize workspace folders, which is how the custom config files are discovered on LCLS initialization. Related to #483

**Also to note**, it seems the intellij-quarkus plugin has made changes to enable workspace folders, and also enable the LSP protocol for `workspace/didChangeWatchedFiles`, which may replace these changes at some point in the future.

### CODE REVIEW

File: src/main/java/io/openliberty/tools/intellij/liberty/lsp/LibertyCustomConfigListener.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
+package io.openliberty.tools.intellij.liberty.lsp;
```
Comment (cherylking) [inline src/main/java/io/openliberty/tools/intellij/liberty/lsp/LibertyCustomConfigListener.java:1]: add a copyright statement

File: src/main/java/io/openliberty/tools/intellij/liberty/lsp/LibertyCustomConfigManager.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
+package io.openliberty.tools.intellij.liberty.lsp;
```
Comment (cherylking) [inline src/main/java/io/openliberty/tools/intellij/liberty/lsp/LibertyCustomConfigManager.java:1]: copyright

File: src/main/java/io/openliberty/tools/intellij/liberty/lsp/LibertyCustomConfigManager.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    @Override
+    public void dispose() {
+
+    }
+
+    public interface Listener {
+        void processConfigXml(List<String> uris);
+    }
+
+    public static LibertyCustomConfigManager getInstance(Project project) {
```
Comment (turkeylurkey) [inline src/main/java/io/openliberty/tools/intellij/liberty/lsp/LibertyCustomConfigManager.java:41]: The similar class LanguageServiceAccessor declares the parameter as `@NotNull Project project`

**Review by turkeylurkey — APPROVED** [review]
Seems to work!

File: src/main/java/io/openliberty/tools/intellij/lsp4mp/lsp4ij/LanguageServerWrapper.java [hunk 4]
```diff
@@ -363,11 +363,11 @@ private void logMessage(Message message) {
...
         } else if (LOGGER.isDebugEnabled()) {
-            LOGGER.info(message.getClass().getSimpleName() + '\n' + message.toString());
+            LOGGER.warn(message.getClass().getSimpleName() + ": " + this.serverDefinition.id + '\n' + message.toString());
         }
-//        else {
-//            LOGGER.warn(message.getClass().getSimpleName() + '\n' + message.toString());
-//        }
+        // else {
+        //     LOGGER.warn(message.getClass().getSimpleName() + ": " + this.serverDefinition.id + '\n' + message.toString());
+        // }
```
Comment (TrevCraw) [inline src/main/java/io/openliberty/tools/intellij/lsp4mp/lsp4ij/LanguageServerWrapper.java:370]: do we still need to keep this commented "else" block?

File: src/main/java/io/openliberty/tools/intellij/liberty/lsp/LibertyCustomConfigManager.java [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+    private final static Logger LOGGER = Logger.getInstance(LibertyCustomConfigManager.class);
+
+    private final Project project;
+    private final MessageBusConnection appConnection;
+    private final LibertyCustomConfigListener listener;
+
+    @Override
+    public void dispose() {
+
```
Comment (TrevCraw) [inline src/main/java/io/openliberty/tools/intellij/liberty/lsp/LibertyCustomConfigManager.java:35]: Are we safe to not do any cleanup on dispose()?

**Review by TrevCraw — APPROVED** [review]

**Review by TrevCraw — APPROVED** [review]

**Pull request merged**